### PR TITLE
feat(ui): Swap `<Role>` for `useRole`

### PR DESCRIPTION
**Pull request by scttcper**
Replaces the `<Role>` component with a `useRole` hook that takes the key on the organization to use.

Previously you would get the result as a child render function.
```
<Role role={organization.attachmentsRole}>
 {hasRole => ...}
</Role>
```

with useRole
```
const {hasRole} = useRole({role: 'attachmentsRole'});
```

### CODE REVIEW

File: static/app/components/events/eventAttachmentActions.tsx [hunk 1]
```diff
@@ -26,65 +26,62 @@ function EventAttachmentActions({
...
-              disabled={!hasAttachmentRole}
-              title={
-                hasAttachmentRole
-                  ? t('Delete')
-                  : t('Insufficient permissions to delete attachments')
-              }
-            />
-          </Confirm>
-        </ButtonBar>
+    <ButtonBar gap={1}>
```
Comment (scttcper) [inline static/app/components/events/eventAttachmentActions.tsx:34]: mostly whitespace changes

File: static/app/components/acl/role.tsx [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-type RoleRenderProps = {
-  hasRole: boolean;
-};
-
-type ChildrenRenderFn = (props: RoleRenderProps) => React.ReactElement | null;
-
-function checkUserRole(user: User, organization: Organization, role: RoleProps['role']) {
-  if (!user) {
-    return false;
-  }
```
Comment (scttcper) [inline static/app/components/acl/role.tsx:19]: I don't know why user was here? It just checks that user exists? Was that important?

File: static/app/components/acl/useRole.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+interface UseRoleResult {
+  hasRole: boolean;
+  /**
+   * The required role ('member', 'admin') from the organization object.
+   */
+  roleRequired: string;
+}
+
+export function useRole(options: UseRoleOptions): UseRoleResult {
```
Comment (ryan953) [inline static/app/components/acl/useRole.tsx:43]: hooks for all the things!

File: static/app/components/acl/role.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/components/acl/role.tsx:19]: Yeah I don't see a situation where you could render this and not have a user available

**Review by malwilley — APPROVED** [review]

**Pull request merged**